aws/resource_aws_ssm_parameter.go

```diff
@@ -87,6 +103,48 @@ func resourceAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("type", param.Type)
 	d.Set("value", param.Value)
 
+	describeParamsInput := &ssm.DescribeParametersInput{
+		Filters: []*ssm.ParametersFilter{
+			&ssm.ParametersFilter{
+				Key:    aws.String("Name"),
+				Values: []*string{aws.String(d.Get("name").(string))},
+			},
+		},
+	}
+	detailedParameters := []*ssm.ParameterMetadata{}
+	err = ssmconn.DescribeParametersPages(describeParamsInput,
+		func(page *ssm.DescribeParametersOutput, lastPage bool) bool {
+			detailedParameters = append(detailedParameters, page.Parameters...)
+			return !lastPage
+		})
+	if err != nil {
+		return errwrap.Wrapf("[ERROR] Error describing SSM parameter: {{err}}", err)
+	}
+	if len(detailedParameters) == 0 {
+		log.Printf("[WARN] SSM Param %q not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	detail := detailedParameters[0]
+	if detail.Description != nil {
+		// Trailing spaces are not considered as a difference
```

Comment:
Does this still apply? If so, looks like we should move this to a `DiffSuppressFunc` on the attribute to make the behavior more clear

Reply:
This actually was a hack due to the API bug which is apparently resolved. Removing.